### PR TITLE
11. GET /api/reviews (queries)

### PR DESCRIPTION
**Pull request by MRombi**
add query for sort_by, order and category(filter) with tests
Added checkUsernameById and tests to check if the username exists when posting comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -190,6 +190,63 @@ describe("GET: /api/reviews", () => {
...
+      });
+  });
+  test("200: returns with all the reviews with same category", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity")
+      .expect(200)
+      .then(({ body }) => {
+        body.reviews.forEach((review) => {
+          expect(review.category).toBe("dexterity");
+        });
```
Comment (douglashellowell) [inline __tests__/app.test.js:223]: remember to include an assertion that the array (`reviews`) is not empty!

If it was empty your `forEach` would not invoke it's function and your jest test would automatically pass! 😱

File: __tests__/app.test.js [hunk 2]
```diff
@@ -200,6 +257,38 @@ describe("GET: /api/reviews", () => {
         });
     });
   });
+  test("400: bad request if method is wrong", () => {
+    return request(app)
+      .get("/api/reviews/?test=euro game")
```
Comment (douglashellowell) [inline __tests__/app.test.js:262]: no need for the `/` at the end of this path

```diff
-      .get("/api/reviews/?test=euro game")
+      .get("/api/reviews?test=euro game")
```

File: db/app/controllers/comments.c.js [hunk 3]
```diff
@@ -23,14 +24,15 @@ exports.getCommentByReviewId = (req, res, next) => {
 
 exports.postCommentByReviewId = (req, res, next) => {
   const { review_id } = req.params;
+  const { username } = req.body;
   const allPromises = Promise.all([
     selectReviewById(review_id),
+    checkUsernameById(review_id, username),
```
Comment (douglashellowell) [inline db/app/controllers/comments.c.js:30]: this `checkUsernameById` check isn't strictly necassary - the `insertCommentByReviewId` model would have thrown an error if the username didn't exist already

**Comment by douglashellowell** [comment]
Forgot to mention!

I've just noticed you've put all your code in `/db`!

You should only put the database-related files in here

File: db/app/models/reviews.m.js [hunk 4]
```diff
@@ -86,26 +81,88 @@ exports.updateReviewById = (id, votes) => {
...
+  ]);
+  let checkSortOptions = sortByOptions.has(sort_by);
+
+  if (checkSortOptions === false) {
+    return Promise.reject({
+      status: 400,
+      message: "Bad request, incorrect sort_by",
+    });
+  } else {
+    searchSort_by = sort_by;
```
Comment (douglashellowell) [inline db/app/models/reviews.m.js:120]: this variable looks like its not being used anymore - is it left over from before the refactor?

File: db/app/models/comments.m.js [hunk 5]
```diff
@@ -54,13 +49,13 @@ exports.insertCommentByReviewId = (id, comments) => {
   (body, review_id, author)
   VALUES
   ($1, $2, $3)
-  RETURNING body;
+  RETURNING author, body;
```
[the remark this answers is not in the document]
Comment (douglashellowell) [reply db/app/models/comments.m.js:52]: 👆 my previous comment still needs addressing here

**Review by douglashellowell — CHANGES_REQUESTED** [review]
nice work!

There is still a fix needed in your `insertCommentByReviewId` model (see my comment above)

---

still need to `200` test on a reviews about a category that does exist but doesn't have any reviews related to it

---

**Review by Angaca — APPROVED** [review]
Looks very good!

**Pull request merged**